explainer.md

```diff
@@ -0,0 +1,255 @@
+    .animated {
+      position: relative;
+      top: 100px;
+      width: 200px;
+      height: 200px;
+      background-color: red;
+    }
+    
+    @keyframes fadeIn {
+      0% {opacity: 0;}
+      100% {opacity: 1;}
+    }
+    
+    @keyframes fadeOut {
+      0% {opacity: 1;}
+      100% {opacity: 0;}
+    }
+    
+    ::shared-new(#root), ::shared-new(#shared-header) {
+      animation: fadeIn 1s;
+    }
+
+    ::shared-old(#root), ::shared-old(#shared-header) {
+      animation: fadeOut 1s;
+    }
+    
+    ::shared-container(#shared-header) {
+      transition: all 1s ease-in;
+    }
+  </style>
+</head>
+<body>
+  <div class="animated" shared-id="shared-header" id="header"></div>
+</body>
+</html>
+```
+
+The steps taken by the browser during the transition are as follows.
+
+1. When a navigation is initiated on the old Document, create the following pseudo elements in the top layer :
+    a. A container and child replaced element for each element with shared-id attribute. These are identified via ::shared-container(#shared-id) and ::shared-old(#shared-id) respectively.
+    b. A replaced element for the root/html element identified via ::shared-old(#root).
+
+2. Apply the following UA stylesheet to the pseudo elements on the old page :
+```
+::shared-old(#root), ::shared-container(#shared-header) {
+  position: fixed;
+  top: 0px;
+  left: 0px;
+}
+
+::shared-container(#shared-header) {
+  // Container sized to the element's border-box size.
+  width: 100px;
+  height: 100px;
+  
+  // A transform mapping the element to its quad in viewport space.
+  transform: translate(0px, 0px);
+}
+
+::shared-old(#shared-header) {
+  position: absolute;
+  width: 100%;
+  height: 100%;
+  content: element(#header);
+}
+
+::shared-old(#root) {
+  width: 100vw;
+  height: 100vh;
+  // The output of element() function on the root element.
+  content: element(html);
+}
+```
+
+3. Save the output of the element() function for each pseudo element referenced above with the computed size and transform applied to container elements. Then navigate to the new page leaving the last rendered pixels of the old page on screen.
+
+4. When the new page loads, suppress rendering until resources required for first render have been fetched. This state is currently driven by browser heuristics but the eventual goal is to give deterministic control to developers using a [renderblocking](https://github.com/whatwg/html/issues/7131) attribute.
+
+5. Once the page is ready for first render, create the following pseudo elements in the top layer. This is done until the end of rAF callbacks on the page's first rendering lifecycle update[^1] :
+    a. A container and child replaced element for each shared element on the old page using state saved in step 3.
+    b. A container and child replaced element for each element with shared-id attribute on the new page. Only the child element is created if a container with the matching id was already created in step a. above.
```

Comment:
Rephrase second sentence to something like "the container is reused if already present"

Reply:
Done.